Zend/zend_hash.c

```diff
@@ -2347,11 +2347,16 @@ static zend_always_inline bool zend_array_dup_element(HashTable *source, HashTab
 // We need to duplicate iterators to be able to search through all copy-on-write copies to find the actually iterated HashTable and position back
 static void zend_array_dup_ht_iterators(HashTable *source, HashTable *target) {
 	HashTableIterator *iter = EG(ht_iterators);
-	HashTableIterator *end  = iter + EG(ht_iterators_used);
+	uint32_t nr_iterators_used = EG(ht_iterators_used);
+	HashTableIterator *end = iter + nr_iterators_used;
 
 	while (iter != end) {
 		if (iter->ht == source) {
+			size_t iter_index = iter - EG(ht_iterators);
```

Comment:
Alternative idea is to use indices in the while loop, but then we have to fetch `EG(ht_iterators)` always in the loop body which seemed more expensive.

Reply:
I think indices are going to be better.

Reply:
Ok I have pushed a commit to do that.